### PR TITLE
[ENH] General Fixes

### LINKED ISSUE #812: Enhancements/Fixes/Suggestions for pivot_longer/pivot_wider
-  [X] use the `pivot` function instead of `stack/unstack`, for `pivot_wider`, since the minimum version is pandas 1.1 (the `pivot` function in versions prior to pandas 1.1 did not support lists in the `index`/`columns` parameters)
- ~~not sure if it is necessary to have an `aggfunc` parameter in `pivot_wider`. Maybe deprecate?~~
- ~~use `stack` where possible for `pivot_longer` - it should offer a simpler code and more maintainable~~
- ~~stick to single index columns for `pivot_longer` - more maintainable (current code looks spaghetti and brittle); any multiIndex column transformations can be done by the user with `stack` or `melt`. if accepted, this means deprecating the `col_level` parameter. also remove it from the `_select_columns` function.~~
- [X] Use the `ignore_index` capability from pandas `melt` directly, instead of the custom function created (since pandas minimum version in pyjanitor is now 1.1)
- [X] For `select_columns` return tuples as tuples (allows column names that are tuples to be selected)
- [X] remove `fill_value` from `complete`; user can always achieve this with better results with `fillna`
- [X] fix errors with groupby in `complete`

### PR DESCRIPTION
**Pull request by samukweku**
# PR Description

Please describe the changes proposed in the pull request:

- Drop `fillna` from `complete` function. Users are better served with `fillna` instead.
- Use `pivot` for `pivot_wider`, since minimum pandas version is 1.1
- Fix errors with the `by` parameter in the `complete` function.
- `pivot_longer` uses more functions directly from the Pandas library directly (`ignore_index`)

**This PR resolves #812.**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: janitor/functions.py [hunk 1]
```diff
@@ -5781,10 +5762,10 @@ def pivot_longer(
 @pf.register_dataframe_method
 def pivot_wider(
     df: pd.DataFrame,
-    index: Optional[Union[List, str]] = None,
-    names_from: Optional[Union[List, str]] = None,
-    values_from: Optional[Union[List, str]] = None,
-    names_sort: Optional[bool] = False,
+    index: Optional[Union[Sequence[str], str]] = None,
+    names_from: Optional[Union[Sequence[str], str]] = None,
+    values_from: Optional[Union[Sequence[str], str]] = None,
```
Comment (ericmjl) [inline janitor/functions.py:5767]: Usually I've seen `Iterable[str]`, rather than `Sequence[str]`. May I ask what's the difference?

**Review by ericmjl — APPROVED** [review]

**Comment by ericmjl** [comment]
This is wonderful, @samukweku! Thanks for the work that went into the PR. I'm pre-approving. Let's see if we can get any more reviews in.

File: janitor/functions.py (hunk 1 above)
Comment (samukweku) [reply janitor/functions.py:5767]: @ericmjl I think I noticed it in pandas pivot do strings; tried to follow their annotation 😁

Comment (ericmjl) [reply janitor/functions.py:5767]: Okie dokes! If the `pandas` dev team does it, I'm sure there's a good reason.

**Comment by ericmjl** [comment]
@samukweku it's been a bit more than a week now, so I think we should merge!

**Comment by samukweku** [comment]
Just noticed a bug in pivot longer; I will push a fix for it today

**Pull request merged**